### DISCUSSION TITLE
How to use Supabase with server + client rendering via NextJS?

### DISCUSSION THREAD

**Discussion by vymao** [supabase/supabase#11377] [Questions]
Hi all, 

I'm wondering if it is currently possible to have Supabase work with both server + client rendering. It seems as though we must generally choose one or the other, but NextJS allows one to define SSG (server-side generated), SSR (server-side rendered), and CSR (client-side rendered) pages. 

For example, the [webapp example](https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs) for NextJS + Supabase seems to indicate that we should `SessionContextProvider`; however, wrapping the entire app in this context provider requires `useState`, which to my understanding will render the entire app as client-side rendered. 

Supabase has a [server-side rendering solution](https://supabase.com/docs/guides/auth/server-side-rendering), but then this seems to require us to manually handle components/pages that we want to render client-side. 

With the NextJS-13 beta, we can explicitly define server components, and it seems Supabase has a [beta solution](https://supabase.com/docs/guides/auth/auth-helpers/nextjs-server-components#authentication) too. However, I was wondering if it is currently possible to do this without the beta?

**Comment by GaryAustin1** [comment]
Many/Most server side users are using Auth-Helpers (your 3rd mention) as the way to deal with this.
